pkg/snapstore/abs_snapstore.go

```diff
@@ -53,55 +86,70 @@ type absCredentials struct {
 func NewABSSnapStore(config *brtypes.SnapstoreConfig) (*ABSSnapStore, error) {
 	absCreds, err := getCredentials(getEnvPrefixString(config.IsSource))
 	if err != nil {
-		return nil, fmt.Errorf("failed to get sharedKeyCredential: %v", err)
+		return nil, fmt.Errorf("failed to get credentials: %w", err)
 	}
 
-	sharedKeyCredential, err := azblob.NewSharedKeyCredential(absCreds.StorageAccount, absCreds.StorageKey)
+	sharedKeyCredential, err := container.NewSharedKeyCredential(absCreds.StorageAccount, absCreds.StorageKey)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create shared key sharedKeyCredential: %v", err)
+		return nil, fmt.Errorf("failed to create sharedKeyCredential: %w", err)
 	}
 
-	pipeline := azblob.NewPipeline(sharedKeyCredential, azblob.PipelineOptions{
-		Retry: azblob.RetryOptions{
-			TryTimeout: downloadTimeout,
-		}})
-
 	domain := brtypes.AzureBlobStorageGlobalDomain
 	if absCreds.Domain != nil {
 		domain = *absCreds.Domain
 	}
 
-	blobURL, err := ConstructBlobServiceURL(absCreds.StorageAccount, domain)
+	blobServiceURL, err := ConstructBlobServiceURL(absCreds.StorageAccount, domain)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to construct the blob service URL with error: %w", err)
+	}
+	containerURL := fmt.Sprintf("%s/%s", blobServiceURL, config.Container)
+
+	client, err := container.NewClientWithSharedKeyCredential(containerURL, sharedKeyCredential, &container.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Retry: policy.RetryOptions{
+				TryTimeout: downloadTimeout,
+			},
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create client with shared key credential with error: %w", err)
 	}
 
-	serviceURL := azblob.NewServiceURL(*blobURL, pipeline)
-	containerURL := serviceURL.NewContainerURL(config.Container)
+	// Check if the ABS container exists (moved over from client constructor function)
```

Comment:
I didn't get this comment, can you please explain where you are checking the `ABS container exists` or it is a todo ?

Reply:
As can be seen in: https://github.com/gardener/etcd-backup-restore/blob/76664bff40c39f209968bde06bf711424202ed3a/pkg/snapstore/abs_snapstore.go#L204-L213
when the Snapstore struct is created from the client, a validation is performed to check if the container that is passed as configuration to etcd-backup-restore exists or not.

This was moved from `GetABSSnapstoreFromClient` to `NewABSSnapStore`. This move makes it easier for testing when `NewABSSnapStoreFromClient` is used to create a mock client.

Reply:
ok, make sense

Reply:
I will be removing this comment since it is not necessary anymore.